File: CyberCloudBackend/speakly/src/main/java/tn/esprit/entity/BlogPost.java
package tn.esprit.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity

public class BlogPost implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long postId;

    String title;
    String content;

    @Enumerated(EnumType.STRING)
    Reaction reaction;

    @OneToMany(mappedBy = "blogPost", cascade = CascadeType.ALL, orphanRemoval = true)
    @ToString.Exclude
    List<BlogComment> comments;

    @OneToMany(mappedBy = "blogPost", cascade = CascadeType.ALL, orphanRemoval = true)
    @ToString.Exclude

    List<Image> images;

    @ManyToOne
    @JoinColumn(name = "user_id")

    User user;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(nullable = false, updatable = false)
    Date createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = new Date();
    }
}

File: src/app/interfaces/BlogPost.ts
export interface BlogComment {
  commentId: number;
  userId: number;
  content: string;
  createdAt: Date;
}

export enum Reaction {
  LIKE = 'LIKE',
  HAHA = 'HAHA',
  SAD = 'SAD',
  LOVE = 'LOVE',
  ANGRY = 'ANGRY',
  DISLIKE = 'DISLIKE'
}

export interface Image {
  imageId: number;
  url: string;
  description: string;
  orderIndex: number;
}

export interface BlogPost {
  postId: number;
  userId: number | null;
  title: string;
  content: string;
  reaction: Reaction | null;
  comments?: BlogComment[];
  images?: Image[];
  createdAt: Date;
  user:{
    firstName: string;
    lastName: string;
    id?: number;
  }
}
  
  
    